### PR TITLE
Add file migration system requirements

### PR DESCRIPTION
**Pull request by bergice**
Add system requirements to file migration so people know how to approach it depending on how many files they need to migrate.

See https://github.com/silverstripeltd/open-sourcerers/issues/45 for more information on how these estimates were obtained.

### CODE REVIEW

**Comment by robbieaverill** [comment]
For those without access to the private issue linked:

>* Check web request operation (via dev/tasks/MigrateFileTask), capped at 256MB mem and 30s exec
  **SSP Virtual Stack: 150 File DataObjects upgraded, Execution time: 30 seconds, Peak memory usage: 6 MB**
>* Check queuedjobs operation (by manually adding this task as a one off execution), capped at 256MB mem and 120s exec
  **SSP Virtual Stack: 500 File DataObjects upgraded, Execution time: 120 seconds, Peak memory usage: 8 MB**
>* Check manual CLI run, capped at 2GB mem and no exec limit
  **Vagrant silverstripeltd/dev-ssp box: 10000 File DataObjects upgraded, Execution time: 10000 seconds, Peak memory usage: 916 MB**

File: docs/en/02_Developer_Guides/14_Files/05_File_Migration.md [hunk 1]
```diff
@@ -100,3 +100,16 @@ SilverStripe\Core\Injector\Injector:
...
 
 You can also set this to `0` to disable the limit.
+
+## System Requirements
+
+The approach to running your file migration depends on your system and how many files you are migrating.
+
+Please use the following estimates to decide how you will run your file migration:
+
+| Number of files | Method | Expected Execution Time | Expected Memory Usage |
```
Comment (maxime-rainville) [inline docs/en/02_Developer_Guides/14_Files/05_File_Migration.md]:
```suggestion
| Number of files | Method | Expected Execution Time | Approximate Memory Usage |
```

I think _approximate_ is a better word in this context. I think it's clearer that you will get different results.

File: docs/en/02_Developer_Guides/14_Files/05_File_Migration.md [hunk 2]
```diff
@@ -100,3 +100,16 @@ SilverStripe\Core\Injector\Injector:
...
+| --- | --- | --- | --- |
+| < 150 | Web Request | 30 seconds | 6 MB |
+| < 500 | Queued Job | 120 seconds | 8 MB |
+| < 10000 | Command Line | 10000 seconds | 916 MB |
```
Comment (maxime-rainville) [inline docs/en/02_Developer_Guides/14_Files/05_File_Migration.md]:
```suggestion
| < 10000 | Command Line | 10000 seconds | 950 MB |
```

I would go with a round number here. 916MB is bit too specific in this context.

File: docs/en/02_Developer_Guides/14_Files/05_File_Migration.md [hunk 3]
```diff
@@ -100,3 +100,16 @@ SilverStripe\Core\Injector\Injector:
...
+| 10000+ | Command Line or contact support | n/a | n/a |
```
[the remark this answers is not in the document]
Comment (bergice) [reply docs/en/02_Developer_Guides/14_Files/05_File_Migration.md]: These estimates were primarily for SSP users.

How would you like this to be presented?

[the remark this answers is not in the document]
Comment (maxime-rainville) [reply docs/en/02_Developer_Guides/14_Files/05_File_Migration.md]: I think the important thing is the order of magnitude, not necessarily how close the estimate will be to reality. Basically, if you are importing 10K file, expect this to take hours. Whether it takes 5 hours or 3 hours is not really that important.

[the remark this answers is not in the document]
Comment (maxime-rainville) [reply docs/en/02_Developer_Guides/14_Files/05_File_Migration.md]: Made an executive decision and added my own blurb.

**Pull request merged**